### DISCUSSION TITLE
Shapes for Plot?

### DISCUSSION THREAD

**Discussion by erflot** [emilk/egui#224] [Q&A]
Hello,
the Plot widget is nice. 
I tried to add a Vec of filled Rect for a simple histogram.
However, simply adding
shapes: Vec<Shape> to struct Plot fails 
because '==' and '!=' cannot be applied to egui::Shape.
What is the recommended way?
Thank you very much

**Comment by emilk** [comment]
Not exactly sure what the question is, but I fixed the `==` problem for `Shape` in https://github.com/emilk/egui/commit/30885b85f92e2cedebe67aab7e6da3585cd6ec57 so maybe that helps :)

**Comment by erflot** [comment]
Thanks, it compiles with these changes.

I want to use the Plot widget for 4 simple charts:
- histogram
- boxplot
- scatterplot
- heatmap

All these need filled shapes like rect, circle, path, and text, not only curves.
For a test I therefore added `shapes: Vec<Shape>` to struct Plot in a local copy of widgets/plot.rs.
It would be great, if the Plot widget could take a `Vec<Shape>` and draw it like it does for `curves: Vec<Curve>`.
However, this task has low urgency. 
Thank you very much again.

```
#[derive(Clone, PartialEq)]
pub struct Plot {
    next_auto_color_idx: usize,
    shapes: Vec<Shape>,
    curves: Vec<Curve>,
    ...
```

**Reply by emilk** [reply]
The problem with that is that the plot supports non-uniform scaling, and that is difficult for things like circles and text.

This will require a bit of design-work. For instance: when zooming in, should circle become larger? I'm guessing in some cases yes, but maybe not in other cases.

There is some further discussion in https://github.com/emilk/egui/issues/155

You may also want to check out https://github.com/EmbersArc/eplot

I am open to adding more shapes to `egui::plot`, but I want to make sure we think it through first. If you have some design ideas, please open an issue for `egui`!

**Comment by erflot** [comment]
Thanks for pointing me to https://github.com/EmbersArc/eplot.
This seems to have all the items needed, except rotated text as already mentioned by the author.
I will try to put some parts in a widget, because I need static plots only, i.e. no  zoom, no translation, no hover, ...
